Various improvements & tweaks

WizardCM opened the pull request:
Since making the changes to browser, you mentioned that they're split on purpose. You can uncomment the browser stuff and change the corepath to Jim's default if that makes things easier.


* Allow defining build directory, core path & browser path with defaults
* Use app name variable in more places
* Use existing OBS icon from cmake
* Don't grab browser files separately
* Use consistent CompanyName, FileDescription & FileVersion
* Added Readme link
* Link to Quickstart guide
* Improved wording & naming in Uninstaller

Commit:
media-playback: Unbuffered Media Source

When using an IP camera on a local network, we wanted to minimize
delay. In order to achieve minimum delay, we allowed Media Source to
set BufferingMB to 0, and when it is 0, also enable AVFMT_FLAG_NOBUFFER
in the AVFormatContext flags.

Chiitoo committed:
deps/glad: Fix build with GCC-10

GCC-10 defaults to '-fno-common' [1], which triggers issues with
defining global variables multiple times.  To fix the build, use
'extern' to turn the first definition of 'gladGetProcAddressPtr'
into a declaration.

1. https://gcc.gnu.org/bugzilla/show_bug.cgi?id=85678

Closes https://github.com/obsproject/obs-studio/issues/2828

WizardCM committed:
UI: Match Windows taskbar state to tray icon

This adds both a status icon, and changes the color
of the taskbar button itself.

Bennik2000 committed:
UI: Fix wrong path in the crash message dialog

Closes obsproject/obs-studio#2290

hselasky committed:
libobs: Implement and use better scaling function for 64-bit integers

As os_gettime_ns() gets large the current scaling methods, mostly by casting
to uint64_t, may lead to numerical overflows. Sweep the code and use
util_mul_div64() where applicable.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>

WizardCM committed:
UI: Don't clip meters when resizing with no input

jpark37 committed:
obs-filters: Implement CUBE LUT domain properly

Previously assuming zero to one instead of respecting values.

jpark37 committed:
obs-filters: 3D LUT tetrahedral interpolation

As requested by Adobe CUBE spec.

jpark37 committed:
libobs-opengl: Support 3D texelFetch

pkviet committed:
obs-ffmpeg: Enable mpegts network URL for ffmpeg-mux

pkviet committed:
obs-ffmpeg: Use obs-ffmpeg-mux for mpegts network output

This replaces the ffmpeg-encoded-output which had serious issues
(missing headers, muxing issues with non compliant mpegts streams) with
an output grafted from obs-ffmpeg-mux.

cg2121 committed:
UI: Use standard resolutions with auto-config

pkviet committed:
UI: Swap to new srt output

Redirect non rtmp streaming to ffmpeg-mux instead of previous
ffmpeg-encoded-output which has been removed.

Commit:
UI: Remove language region

jpark37 committed:
win-capture: Use VkAllocationCallbacks

jpark37 committed:
win-capture: Cleaner COM usage

Remove hard-coded GUIDs, call correct functions, and clean up
unnecessary casts.

jpark37 committed:
win-capture: Verify VK_KHR_external_memory_win32 support

Good practice to verify necessary extensions exist. We'll probably need
keyed mutex extension in the future to support Intel.

dgcampea committed:
obs-ffmpeg: Expose VAAPI profile choices

Allow choosing Constrained Baseline, Main and High.

dgcampea committed:
obs-ffmpeg: Rename and add more VAAPI levels

The Auto level lets the encoder decide which level to use.
Furthermore, some devices allow levels beyond 4.2. Therefore,
we add all levels allowed by H.264. This is of importance,
since using a resolution not supported by the specified level
can trigger undefined behavior on the hardware/kernel.

kkartaltepe committed:
UI: Install public headers for frontend-api

Previously the api header was not targeted for installation and may have
been missing for third party users.

kkartaltepe committed:
cmake: Fix warnings and normalize variables/errors

As of 3.17 using find_package_handle_standard_args checks that the name
of the FindXXX file and the first argument are the same case.

Some modules used non-standard variables or the old singular variables
instead of plurals. This normalizes variable usage to the new-style.

Some CMakeLists.txt did custom error checking instead of propagating
find_package errors. These were changes to call find_package with
REQUIRED or without QUIET where needed and shortens the custom status
messages. This helps users who want to enable that functionality see
what precisely wasnt found.

kkartaltepe committed:
cmake: Fix warnings and normalize variables/errors

As of 3.17 using find_package_handle_standard_args checks that the name
of the FindXXX file and the first argument are the same case.

Some modules used non-standard variables or the old singular variables
instead of plurals. This normalizes variable usage to the new-style.

Some CMakeLists.txt did custom error checking instead of propagating
find_package errors. These were changes to call find_package with
REQUIRED or without QUIET where needed and shortens the custom status
messages. This helps users who want to enable that functionality see
what precisely wasnt found.

khng300 committed:
plugins: Add oss-audio plugin

This implements OSS audio input capturing support for OSS-capable OSes.
FreeBSD and DragonFly (not yet tested on) supports are added as a
starting point.

univrsal committed:
vlc-video: Allow metadata retrieval through proc_handler

exeldro committed:
libobs: Don't allow duplicating scene sources

dgeibi committed:
rtmp-services: Add Nimo TV

jp9000 committed:
libobs: Update version to 25.0.8

cg2121 committed:
obs-plugins: Check if sources are showing for media hotkeys

ciriousjoker committed:
rtmp-services: Update Switchboard Live servers

Closes obsproject/obs-studio#2813

cg2121 committed:
obs-scripting: Expose platform functions to scripts

nickosmoi committed:
rtmp-services: Add Xlovecam.com streaming service

WizardCM committed:
UI: Update installer with link to release notes and more

* Use app name variable in more places
* Use existing OBS icon from cmake
* Use consistent CompanyName, FileDescription & FileVersion
* Link to Quickstart guide
* Improved wording & naming in Uninstaller

h-o-sein committed:
rtmp-services: Update GameTips.TV

henke37 committed:
libobs: Use SetThreadDescription if possible

It's the better way to declare thread names, but you need Windows 10,
version 1607 to use it.

RytoEX committed:
obs-ffmpeg: Update error message in process_packet

The error message in the process_packet function was prefixed with
"receive_audio" because it was previously from code in the encode_audio
function which was called from the receive_audio function. This is just
a string change to avoid being misled to believe that the error is
always audio related.

notr1ch committed:
Merge pull request #2793 from NiMO-TV/rtmp-nimo-1

rtmp-services: Add Nimo TV

dodgepong committed:
Merge pull request #2833 from RytoEX/update-process-packet-error

obs-ffmpeg: Update error message in process_packet

RytoEX committed:
libobs: Add Windows 10 release version to crash log

Follow-up to eadb96f.

zhk committed:
libobs: Fix unnecessary duplication

Original code makes duplicated item and bad memory

Closes obsproject/obs-studio#2837

jpark37 committed:
libobs-opengl: Lock Mac parent context during present

Parent context lock keeps GL commands serialized.

Fixes race that causes crash when resizing multiview.

jp9000 committed:
UI: Remove unused variable

Closes obsproject/obs-studio#2843

jp9000 committed:
Merge pull request #2854 from jpark37/mac-present-lock

libobs-opengl: Lock Mac parent context during present

jp9000 committed:
Merge pull request #2835 from RytoEX/crash-log-win-release-id

libobs: Add Windows 10 release version to crash log

jp9000 committed:
Merge pull request #2830 from h-o-sein/master

rtmp-services: Update GameTips.TV

jp9000 committed:
Merge pull request #2817 from cg2121/script-functions

obs-scripting: Expose platform functions to scripts

jp9000 committed:
Merge pull request #2814 from cg2121/media-hotkeys

obs-plugins: Check if sources are showing for media hotkeys

jp9000 committed:
Merge pull request #2831 from henke37/SetThreadDescription

libobs: Additionally use SetThreadDescription if possible

jp9000 committed:
Merge pull request #2818 from nickosmoi/xlove_20200427

rtmp-services: Add Xlovecam.com streaming service

jp9000 committed:
Merge pull request #2789 from univrsal/vlc-metadata-procs

vlc-video: Add procedure to receive media metadata

jp9000 committed:
Merge pull request #2864 from Chiitoo/gcc-10

deps/glad: Fix build with GCC-10

jp9000 committed:
Merge pull request #2738 from exeldro/scene_not_duplicate

libobs: Don't allow duplicating scene sources

notr1ch committed:
UI/updater: Fix launching OBS as admin post-update

A long standing lack of synchronization on the elevated process handle
made the non-elevated process always exit, so the updater would assume
it was started elevated and launch OBS from the elevated context.

Also applies the altered working directory to the non-elevated code
path so it can find the OBS executable properly.

ePirat committed:
obs-ffmpeg: Fix AVFrame handling in FFmpeg output

Fix an issue in the way AVFrames were handled in the FFmpeg
encoder plugin, which could lead to tearing in the encoded
video due to data races in the AVFrame and AVBuffer.

This is fixed by calling av_frame_make_writable which ensures
the frame and its associated buffer are writable. If its not,
it will copy the AVFrame, create a new AVBuffer for it and
decrease the refcount of the old AVFrame and AVBuffer.
This way OBS always ends up with a usable buffer to write into
which is not still used by the encoder while avoiding a copy
when unnecessary.

PatTheMav committed:
CI: Update Github Actions with caching and macOS improvements

jpark37 committed:
libobs-winrt: win-capture: Detect GraphicsCaptureItem closure

Make WGC window capture recover from GraphicsCaptureItem closure, which
can occur when following links in fullscreen Chrome for example.

jp9000 committed:
Merge pull request #2880 from jpark37/wgc-handle-item-closed

libobs-winrt: win-capture: Detect GraphicsCaptureItem closure

jp9000 committed:
Merge pull request #2877 from PatTheMav/ci-update

CI: Update Github Actions with caching and macOS improvements

jp9000 committed:
Merge pull request #2870 from ePirat/epirat-ffmpeg-frame-tearing-fix

obs-ffmpeg: Fix AVFrame handling in FFmpeg output

jp9000 committed:
Merge pull request #2712 from khng300/freebsd-oss-audio

plugins: Add oss-audio input support

jpark37 committed:
mac-capture: Use resize instead of reserve

Better practice for the tracked size to be nonzero.

This does NOT speed up window capture significantly.

jpark37 committed:
libobs: Fix da_reserve early return logic

Test desired capacity against original capacity instead of size.

jpark37 committed:
libobs/util: Fix POSIX event bugs

Handle spurious wakeups, and ensure mutex unlock.

jp9000 committed:
Merge pull request #2895 from jpark37/mac-resize

mac-capture: Use resize instead of reserve

jp9000 committed:
Merge pull request #2896 from jpark37/reserve-logic-fix

libobs: Fix da_reserve early return logic

jp9000 committed:
Merge pull request #2900 from jpark37/fix-posix-events

libobs/util: Fix POSIX event bugs

jpark37 committed:
libobs-winrt: Fix WGC minimize handling

Illegal CopySubresourceRegion parameters were sometimes computed when
minimizing and restoring WGC window captures, leading to device loss.

Add robust safeties to ensure that doesn't happen.

jpark37 committed:
libobs: Remove log entry for CoInitializeEx pass

Remove entry from the top of the log. Always returns S_FALSE, and that's
unlikely to change.

Fenrirthviti committed:
Merge pull request #2910 from jpark37/unlog-com-success

libobs: Remove log entry for CoInitializeEx pass

jp9000 committed:
UI: Restore theme if settings window exit with [x]

Fixes a bug where if the theme was changed, then wasn't saved, and the
user exits the window with the [x] button, then chooses "No" when asked
to save settings, it would not restore the original theme.

jpark37 committed:
win-capture: Fail on unsupported Vulkan formats

Don't allow unsupported Vulkan formats to fall back to B8G8R8A8.
Probably better to fail completely than do an illegal copy.

Also remove bad conversion for VK_FORMAT_A2R10G10B10_UNORM_PACK32.
Red and blue channels were reversed, and there's no DXGI equivalent.

Addresses #2796. We can do more later if justified.

jp9000 committed:
cmake: Add cmake folders

jpark37 committed:
win-capture: Reset WGC fail flag for new window

WGC will give up on a window that it fails to capture, but that
shouldn't stop it from attempting to capture new windows.

Fixes #2928.

tt2468 committed:
libobs: Return target instead of current in calc_torquef

If the previously calculated torque value is too close to 1.0
(but not considered 1.0), it would be impossible for the TBar to
finish its transition since it returned the old value, causing desync.

dodgepong committed:
Merge pull request #2930 from tt2468/patch-1

libobs: Return target instead of current in calc_torquef

brittneysclark committed:
obs-qsv11: Add latency mode to QSV settings

Simplify UI options by combining LookAhead Depth and Async Depth into
latency mode option. Ultra-low, low, and normal will set these two
encode parameters accordingly.

brittneysclark committed:
obs-qsv11: Enable VDEnc on ICL+

Enable AVC VDEnc (fixed function encode) on ICL+

brittneysclark committed:
obs-qsv11: Simplify UI subjective quality parameters

Combine MMBRC and CQM parameters into a subjective video
enhancement box

jpark37 committed:
win-capture: Ignore cloaked windows

Fixes issue where there are two window handles for minimized Calculator,
and OBS picks the wrong one.

notr1ch committed:
Merge pull request #2929 from jpark37/wgc-fail-invalidate

win-capture: Reset WGC fail flag for new window

notr1ch committed:
Merge pull request #2588 from Vainock/remove-language-region

UI: Remove language region

cg2121 committed:
UI: Hide script tabs if no python settings

RytoEX commented on UI/installer/mp-installer.nsi:
```diff
@@ -11,7 +11,18 @@ ManifestDPIAware true
 !endif
 
-!define APPNAMEANDVERSION "OBS Studio ${SHORTVERSION}"
+!ifndef BUILDIR
+!define BUILDIR "build"
```
Typo:  BUILDIR rather than BUILDDIR.

WizardCM replied:
Hah, oops! I'll get that fixed when I update the PR based on your other suggestions.

WizardCM replied:
After a discussion with Fenrir, I've decided to delay this change to the structure of the files.

RytoEX commented on UI/installer/mp-installer.nsi:
```diff
@@ -267,48 +285,48 @@ Section "OBS Studio" SecCore
-	SetOutPath "$INSTDIR"
-	File /r "new\obs-browser\data"
+	# SetOutPath "$INSTDIR"
+	# File /r "new\obs-browser\data"
```
Is there a reason these are commented out rather than removed?

RytoEX commented on UI/installer/mp-installer.nsi:
```diff
@@ -267,48 +285,48 @@ Section "OBS Studio" SecCore
-	File /r "new\obs-browser\obs-plugins\32bit"
-	SetOutPath "$INSTDIR\bin\32bit"
-!endif
+# !ifdef INSTALL64
+# 	File /r "new\obs-browser\obs-plugins\64bit"
+# 	SetOutPath "$INSTDIR\bin\64bit"
+# !else
+# 	File /r "new\obs-browser\obs-plugins\32bit"
+# 	SetOutPath "$INSTDIR\bin\32bit"
+# !endif
```
Same with these.  Why comment out rather than remove?

WizardCM replied:
As mentioned in the description of the PR, Jim prefers they stick around, though it seemed to work OK without for me.

RytoEX replied:
Would it be acceptable to default to these paths if no BUILDDIR is specified instead of leaving them in a comment?

RytoEX commented on UI/installer/mp-installer.nsi:
```diff
@@ -446,22 +464,25 @@ Section "un.obs-studio Program Files" UninstallSection1
 SectionEnd
 
-Section /o "un.User Settings" UninstallSection2
+Section /o "un.Settings, Scenes, etc" UninstallSection2
```
I'm not sure that "Settings, Scenes, etc" is truly clearer than just "User Settings" here.  If this stays, please at least use "etc." with a period at the end.

WizardCM replied:
We already have users that end up having one of these issues:
1. "I reinstalled OBS but all my scenes are still there, why? I wanted a clean reinstall"
2. "I uninstalled OBS and now all my scenes are missing"

The goal of this change is to make it clearer for both situations.

The reason for why I didn't include the dot on the end is because the text as-is *just* fits, to the pixel, and a dot would likely add a horizontal scrollbar or be cut off.

RytoEX replied:
There already is a horizontal scrollbar for this on my screen because the first item is so long.
![image](https://user-images.githubusercontent.com/624931/82217643-b97c7f00-98e8-11ea-8c0c-37ba4aa7b22f.png)

WizardCM replied:
Updated the strings per your comments.

RytoEX commented on UI/installer/mp-installer.nsi:
```diff
@@ -446,22 +464,25 @@ Section "un.obs-studio Program Files" UninstallSection1
 !insertmacro MUI_UNFUNCTION_DESCRIPTION_BEGIN
 	!insertmacro MUI_DESCRIPTION_TEXT ${UninstallSection1} "Remove the OBS program files."
-	!insertmacro MUI_DESCRIPTION_TEXT ${UninstallSection2} "Removes all settings, plugins, scenes and sources, profiles, log files and other application data."
+	!insertmacro MUI_DESCRIPTION_TEXT ${UninstallSection2} "Removes all settings, scenes and sources, profiles, log files and other application data.\r\n\r\nCANNOT BE UNDONE."
```
Use "This cannot be undone." rather than "Cannot be undone."

RytoEX replied:
Unfortunately, it looks like \r\n does not work here.
![image](https://user-images.githubusercontent.com/624931/82220039-21809480-98ec-11ea-8136-54841d5314ac.png)

RytoEX replied:
Figured this out.  You have to use $\r$\n.

WizardCM replied:
Aha, perfect! Updated. :)

RytoEX commented on UI/installer/mp-installer.nsi:
```diff
@@ -446,22 +464,25 @@ Section "un.obs-studio Program Files" UninstallSection1
 !insertmacro MUI_UNFUNCTION_DESCRIPTION_BEGIN
 	!insertmacro MUI_DESCRIPTION_TEXT ${UninstallSection1} "Remove the OBS program files."
-	!insertmacro MUI_DESCRIPTION_TEXT ${UninstallSection2} "Removes all settings, plugins, scenes and sources, profiles, log files and other application data."
+	!insertmacro MUI_DESCRIPTION_TEXT ${UninstallSection2} "Removes all settings, scenes and sources, profiles, log files and other application data.\r\n\r\nCANNOT BE UNDONE."
```
Use "scenes, sources" rather than "scenes and sources".

RytoEX replied:
Use "log files, and other application data" rather than "log files and other application data".

RytoEX commented on UI/installer/mp-installer.nsi:
```diff
@@ -446,22 +464,25 @@ Section "un.obs-studio Program Files" UninstallSection1
 	RMDir /r "$APPDATA\obs-studio"
 SectionEnd
 
+# These two don't work for the uninstaller
+# !define MUI_UNCOMPONENTSPAGE_TEXT_DESCRIPTION_INFO "Choose which aspects of OBS you want to uninstall."
+# !define MUI_UNCONFIRMPAGE_TEXT_TOP "Check the compoonents you want to uninstall. Keeping Settings unchecked is recommended."
```
Do we need these commented here?  I'm assuming the current text in the uninstaller is auto-generated?
![image](https://user-images.githubusercontent.com/624931/82203188-f76da900-98d0-11ea-8779-5128ed894365.png)

WizardCM replied:
They're commented because they're supposed to work but I couldn't figure out the correct key. Yes, the text you see in your screenshot is auto-generated.

RytoEX replied:
I couldn't find documentation to suggest that MUI_UNCOMPONENTSPAGE_TEXT_DESCRIPTION_INFO exists.  MUI_UNCONFIRMPAGE_TEXT_TOP is documented, however.

RytoEX replied:
I figured this out.  You need to define `MUI_COMPONENTSPAGE_TEXT` settings _before_ the insertmacro for MUI_UNPAGE_COMPONENTS.
https://nsis.sourceforge.io/Docs/Modern%20UI/Readme.html#toggle_pgc
![image](https://user-images.githubusercontent.com/624931/82221543-23e3ee00-98ee-11ea-86e7-221d587563c0.png)

RytoEX replied:
As for MUI_UNPAGE_CONFIRM, you would need to use that with an associated MUI_UNPAGE_CONFIRM insertmacro.
https://nsis.sourceforge.io/Docs/Modern%20UI/Readme.html#toggle_pgu

WizardCM replied:
Thanks for digging into it. I've updated it based on what you found.

WizardCM replied:
Hmm, looks like it still doesn't work. What did I do incorrectly?

RytoEX replied:
I deleted my old work file for this, but I'm pretty sure you need to define MUI_COMPONENTSPAGE_TEXT_TOP not MUI_UNCOMPONENTSPAGE_TEXT_TOP.  Same with MUI_UNCOMPONENTSPAGE_TEXT_COMPLIST.

WizardCM replied:
You would be correct, that fixes it. Pushing the change now.

RytoEX reviewed (changes requested):
I've suggested some changes or asked questions about lines already being changed in this PR.

RytoEX commented:
Frustratingly, using the build directory to find the binaries means I have to copy the 32-bit game capture hook files into the 64-bit build directory (or vice versa) because CMake can't configure both builds into one place.

jp9000 committed:
Merge pull request #2947 from cg2121/script-tabs

UI: Hide script tabs if no python settings

jp9000 committed:
obs-ffmpeg: Move file read error to separate function

Moves the code that generates a file read error message to its own
function for the sake of cleaning things up a bit.

jp9000 committed:
Merge pull request #2665 from pkviet/srtfix3

obs-ffmpeg: Fix srt output

jp9000 committed:
Merge pull request #2506 from Bennik2000/fix_issue_2290

Show accurate path in Crash dialog

DDRBoxman committed:
obs-vst: Compile the vst plugin on linux

DDRBoxman committed:
obs-vst: Fix crash when the blocksize is smaller than frames

jp9000 committed:
Merge pull request #2539 from WizardCM/clipping-visual-fix

UI: Don't clip meters when resizing with no input

jp9000 committed:
Merge pull request #2551 from jpark37/cube-lut-enhance

Implement CUBE LUT domain properly, tetrahedral interpolation for 3D LUTs

jp9000 committed:
Merge pull request #2570 from cg2121/auto-config-res

UI: Use standard resolutions with auto-config

jp9000 committed:
Merge pull request #2589 from jpark37/vulkan-alloc

win-capture: Use VkAllocationCallbacks

jp9000 committed:
Merge pull request #2595 from jpark37/vulkan-cleanup

Vulkan hook COM clean up, and extension check

jp9000 committed:
Merge pull request #2619 from dgcampea/ffmpeg-vaapi

Expose more levels and profiles for FFmpeg VAAPI

jp9000 committed:
Merge pull request #2635 from kkartaltepe/install-headers-1

UI: Install public headers for frontend-api

jp9000 committed:
Merge pull request #2637 from kkartaltepe/cmake-variety-fixes

cmake: Fix warnings and normalize variables/errors

jp9000 committed:
Revert "Merge pull request #2637 from kkartaltepe/cmake-variety-fixes"

This reverts commit d777000a60cb9ccd78b3bdf1fbae3583b55aac0c, reversing
changes made to c9e5f30a78d087e42434813df1c56076049d62c3.

jpark37 committed:
libobs-winrt: Add dispatcher queue API

Suspected necessary for WGC stability.

jpark37 committed:
libobs: WinRT and dispatcher init on graphics thread

Suspected necessary for WGC stability.

jpark37 committed:
libobs: Fix potential truncation warnings

jp9000 committed:
Merge pull request #2908 from jpark37/wgc-minimize

Fix WGC minimize handling

jp9000 committed:
Merge pull request #2921 from jpark37/vulkan-format-fail

win-capture: Fail on unsupported Vulkan formats

jp9000 committed:
Merge pull request #2964 from jpark37/winrt-dispatcher

Set up dispatcher queue for WGC stability

jp9000 committed:
Merge pull request #2967 from jpark37/promotion-warnings

libobs: Fix potential truncation warnings

jp9000 committed:
Merge pull request #2384 from LiamCoal/allow-no-buffering

media-playback: Unbuffered Media Source

jp9000 committed:
Merge pull request #2442 from WizardCM/windows-extras

Windows: Taskbar icon & colour

jp9000 committed:
Merge pull request #2657 from hselasky/improvement

libobs: Implement and use better scaling function for 64-bit integers

kkartaltepe committed:
CI: Require VLC in CI builds, Fix VLC

Previously VLC was downloaded but was skipped since it could not be
found. This sets the VLCPath to ensure it is found.

jp9000 committed:
Merge pull request #2972 from kkartaltepe/cmake-variety-fixes

cmake: Fix warnings and normalize variables/errors

jp9000 committed:
Merge pull request #2971 from kkartaltepe/vlc-force

CI: Require VLC in CI builds, Fix VLC in CI

miawgogo committed:
UI: Fixed case to match what the files are named

The files on disk are lower-case but QSS refrences them in
upper-case. Windows doesn't report errors as the platform isn't case
sensitive. Error messages are generated and icons not to render on
case sensitive platforms such as Linux and Mac

notr1ch committed:
UI: Don't show alpha value for color source

This isn't editable by the user, so it doesn't make sense to show it.

notr1ch committed:
win-capture: Better matching of internal UWP windows

All strings were treated as partial matches before, which caused a
false positive with any executable beginning with "time", notably
affecting the game "Timelie" which used Timelie.exe.

futr committed:
libobs: Fix os_get_executable_path_ptr on Linux

Terminate a non-terminated string generated by readlink()
to pass to dirname() that needs terminated string as a parameter.

jp9000 committed:
Merge pull request #2976 from ioangogo/theme-case-fix

UI: Theme case fix

notr1ch committed:
obs-filters: Misc code cleanups detected by PVS Studio

notr1ch committed:
UI: Set 8x8 as minimum selectable resolution

Since libobs internally rounds to various powers of 2 to support SSE
functions, resolutions that are too low can get rounded to zero. libobs
will fail to startup with a zero resolution, so if a user accidentally
entered anything that rounded to zero OBS would become unusable without
manual fixing by editing the profile .ini.

jp9000 committed:
Merge pull request #2979 from notr1ch/min-res

UI: Set 8 x 8 as minimum selectable resolution

notr1ch committed:
obs-ffmpeg: Show friendly error for NV_ENC_ERR_INVALID_VERSION

kkartaltepe committed:
CI: Improve formatcode.sh efficiency

Batch files into clang-format to prevent needless processes. Now its
twice as fast in real time and uses 4x less resources. The old
implementation is also slower (but pretty close to) running a single
invocation of clang-format on all files.

before
time ./formatcode.sh
real	0m3.860s
user	0m20.975s
sys	0m6.694s

after
time ./formatcode.sh
real	0m1.486s
user	0m5.426s
sys	0m0.203s

DDRBoxman committed:
Merge pull request #2983 from kkartaltepe/formatcode-faster

CI: Improve formatcode.sh efficiency

cg2121 committed:
docs: Remove Doxygen

Fenrirthviti committed:
Merge pull request #3000 from cg2121/remove-doxygen

docs: Remove Doxygen

cg2121 committed:
docs: Remove Doxyfile

dodgepong committed:
Merge pull request #3001 from cg2121/remove-doxygen-config

docs: Remove Doxyfile

exeldro committed:
libobs: Use correct data pointer for hotkey pair

jp9000 committed:
Merge pull request #3017 from exeldro/hotkey_pair_data

libobs: Use correct data pointer for hotkey pair

jpark37 committed:
libobs-winrt: Require Windows 10 SDK 19041

This will prevent compiling without WGC cursor toggle support.

notr1ch committed:
Merge pull request #2977 from futr/fix-non-terminated-str

libobs: Fix os_get_executable_path_ptr on Linux

matclayton committed:
rtmp-services: Add Mixcloud

jp9000 committed:
Merge pull request #2993 from brittneysclark/enable_vdenc

obs-qsv11: Simplify QSV settings in UI and Enable AVC VDEnc on ICL+

notr1ch committed:
image-source: Don't check for changes when hidden

If an image source isn't visible there is no point polling for changes
during the time it's hidden as no one will see them. Should the image
be on unreachable media such as an unavailable network share, this
check would cause the graphics thread to periodically stall trying to
detect changes of a non-visible image source.

If the image is modified while it's hidden, the updated file will now
be loaded as soon as the source is made visible - the update timer
intentionally ticks up while hidden.

Partially fixes some of the behavior described in
https://github.com/obsproject/obs-studio/issues/3007, a full fix
likely requires significant changes to support OS-specific file change
notification APIs.

jp9000 committed:
Revert "Merge pull request #2993 from brittneysclark/enable_vdenc"

This reverts commit 636f0bfdfa81f843a629ec2494e2657dbeab5742, reversing
changes made to b9ea45e4982f4ec26b3f115f0e625289a8f26ae4.

Fenrirthviti committed:
Merge pull request #3022 from matclayton/master

rtmp-services: Add Mixcloud service

Fenrirthviti committed:
Merge pull request #3020 from jpark37/windows-sdk-19041

libobs-winrt: Require Windows 10 SDK 19041

Fenrirthviti committed:
cmake: Update minimum Windows SDK version

notr1ch committed:
UI/updater: Exit with error if elevation failed

In rare cases, "runas" would somehow launch the updater as a non-admin user
again, resulting in a loop where the updater constantly spawns processes.
Fixes https://github.com/obsproject/obs-studio/issues/2984

notr1ch committed:
UI/updater: Fix running updater as different user

If the elevated user was a different user account, the updater would
look in the wrong user's appdata for the manifest and fail.

notr1ch committed:
UI/updater: Use 1 MB static memory for hashing

Allocating a vector for the hundreds of small files and only reading
64k at a time was a bottleneck on systems that were not I/O bound.

jp9000 committed:
Merge pull request #3033 from Fenrirthviti/win-sdk-update

cmake: Update Windows SDK check to 19041

notr1ch committed:
UI: Don't try to load replay buffer hotkey if null

If replay buffer hotkey isn't set, this would cause an error to be
logged due to a null string.

notr1ch committed:
UI: Don't try to create service if missing file

This would cause meaningless "Service '' not found" log entries.

jp9000 committed:
Merge pull request #2820 from Fenrirthviti/nsis-update

UI: Update NSIS installer script

jp9000 committed:
UI: Add 64bit windows checks to installer

Prevents it from writing to non-existent vulkan registry locations on
32bit-only systems.
